fraud/interface.go

```diff
@@ -0,0 +1,43 @@
+package fraud
+
+import (
+	"context"
+)
+
+// ProofUnmarshaler aliases a function that parses data to `Proof`.
+type ProofUnmarshaler func([]byte) (Proof, error)
```

Comment:
What is the rationale of not making this an interface as well then?

Reply:
@liamsi, Why? if we just need a stateless function, not a type. Otherwise, there would be an empty struct like
```go
type unmarshaller struct {}
```

Feels like you are missing the point.